core/shared/src/main/scala/matryoshka/data/option.scala

```diff
@@ -25,6 +25,27 @@ trait OptionInstances {
   implicit def optionBirecursive[A]
       : Birecursive.Aux[Option[A], Const[Option[A], ?]] =
     id.idBirecursive[Option[A]]
+
+  implicit val optionDelayEqual: Delay[Equal, Option] =
+    Delay.fromNT(λ[Equal ~> (Equal ∘ Option)#λ](eq =>
+      Equal.equal {
```

Comment:
I’ll try this, but I _think_ it runs into the same infinite regression thing when trying to resolve, say,  `Mu[Option]`.